ASM/c/debug.c

```diff
@@ -31,71 +38,217 @@ warp_t dungeon_warps[] = {
+    {  0x0, 5, { -428, -880, 961 }, 49050, "Log Spike"},
+    {  0x0, 9, { -614, -1881, -388 }, 34300, "231 Scrubs"},
+};
+
+room_t dc_rooms[] = {
+    {  0x04, 0, { -3, 0, 369 }, 32768, "Entrance"},
+    {  0x04, 2, { -927, 0, -1525 }, 49050, "Bomb Flower Staircase"},
+    {  0x04, 9, { 1392, 531, -316 }, 16384, "Bomb Bag room"},
+    {  0x04, 7, { 7, 76, -2057 }, 32768, "Floor switch room before Boss"},
+};
+
+room_t jabu_rooms[] = {
+    {  0x28, 0, { -3, -226, 369 }, 32768, "Entrance"},
+    {  0x28, 3, { 231, -1113, -3228}, 29920, "B1 Main"},
+    {  0x28, 7, { 6, -340, -4074 }, 32666, "Branching Hallways"},
+};
+
+room_t forest_rooms[] = {
+    {  0x169, 0, { 110, 309, 641 }, 32768, "Entrance"},
+    {  0x169, 2, { 118, 383, -660 }, 32768, "Main Chamber"},
+    {  0x169, 11, { -1447, 383, -1439 }, 49152, "Block Puzzle room"},
+    {  0x169, 12, { -1328, 1228, -3326 }, 16282, "Red Poe"},
+    {  0x169, 13, { 586, 827, -3322 }, 16282, "Blue Poe"},
+    {  0x169, 15, { 1979, 403, -3404 }, 16282, "Checkerboard Ceiling"},
+};
+
+room_t fire_rooms[] = {
+    {  0x165, 0, { 5, 0, 846 }, 32768, "Entrance"},
+    {  0x165, 4, { 2989, 2060, 0 }, 49152, "Shortcut room"},
+    {  0x165, 5, { 2660, 2800, 0 }, 16384, "Boulder maze (down)"},
+    {  0x165, 5, { 1540, 2940, -630 }, 16384, "Boulder maze (up)"},
+    {  0x165, 8, { 1580, 4400, -509 }, 0, "Scarecrow room"},
+    {  0x165, 10, { -640, 2940, 190 }, 49152, "Fire wall maze"},
+    {  0x165, 13, { -2289, 4400, 19 }, 16384, "Hammer room"},
+};
+
+room_t water_rooms[] = {
+    {  0x10, 0, { -184, 780, 759 }, 32768, "Entrance"},
+    {  0x10, 17, { 987, 780, 181 }, 16384, "First ZL water level switch"},
+    {  0x10, 12, { -180, 0, -1870 }, 32768, "Pool of tektites and boulders"},
+    {  0x10, 8, { -1140, 60, -1810 }, 32768, "Dragon room"},
+    {  0x10, 13, { -3124, 1060, -1747 }, 32666, "Dark link room"},
+    {  0x10, 21, { -3113, 380, -4222 }, 4759, "River"},
+};
+
+room_t shadow_rooms[] = {
+    {  0x37, 2, { -254, -63, 594 }, 32768, "Entrance"},
+    {  0x37, 8, { 4164, -983, 1366 }, 49152, "Before huge room"},
+    {  0x37, 16, { 4677, -1143, 2474 }, 32768, "Invisible scythes"},
+    {  0x37, 10, { 1227, -1343, 3853 }, 49152, "Stone umbrella"},
+    {  0x37, 11, { 2477, -1343, 1393 }, 32666, "Invisible spikes"},
+    {  0x37, 21, { 4387, -1363, -1486 }, 32768, "Before boat"},
+    {  0x37, 15, { -3662, -1363, -1585 }, 49152, "Invisible wall maze"},
+};
+
+room_t spirit_rooms[] = {
+    {  0x82, 1, { -768, -50, -1 }, 49152, "After crawlspace (child)"},
+    {  0x82, 0, { 868, -50, 2 }, 16384, "After silver block (adult)"},
+    {  0x82, 15, { 1126, 480, -1281 }, 16384, "Spinning cobra room"},
+    {  0x82, 5, { 692, 480, -838 }, 49152, "Main room"},
+    {  0x82, 18, { 1787, 843, 152 }, 16282, "Armos room"},
+    {  0x82, 25, { 268, 1733, -828 }, 49152, "Ceiling mirror"},
+};
+
+room_t botw_rooms[] = {
+    {  0x98, 0, { 0, -12, 117 }, 32768, "Main room"},
```

Comment:
The exclusion of "Entrance" feels slightly weird, but I guess the main room is just the entrance + a crawlspace